app-editors/atom/atom-1.7.3.ebuild

```diff
@@ -5,81 +5,363 @@
+	https://github.com/atom/scrollbar-style/archive/v${SCROLLBAR_STYLE_V}.tar.gz -> atom-scrollbar-style-${SCROLLBAR_STYLE_V}.tar.gz
+	https://github.com/atom/node-spellchecker/archive/v${NODE_SPELLCHECKER_V}.tar.gz -> atom-node-spellchecker-${NODE_SPELLCHECKER_V}.tar.gz
+
+	https://github.com/nodegit/promisify-node/archive/${PROMISIFY_NODE_V}.tar.gz -> nodegit-promisify-node-${PROMISIFY_NODE_V}.tar.gz
+	https://registry.npmjs.org/nodegit-promise/-/nodegit-promise-${NODEGIT_PROMISE_V}.tgz
+	https://registry.npmjs.org/lodash/-/lodash-${NODE_LODASH_V}.tgz -> node-lodash-${NODE_LODASH_V}.tgz
+	https://github.com/kriskowal/asap/archive/v${ASAP_V}.tar.gz -> node-asap-${ASAP_V}.tar.gz
+	https://github.com/sindresorhus/object-assign/archive/v${OBJECT_ASSIGN_V}.tar.gz -> node-object-assign-${OBJECT_ASSIGN_V}.tar.gz
+	https://github.com/jprichardson/node-fs-extra/archive/${NODE_FS_EXTRA_V}.tar.gz -> node-fs-extra-${NODE_FS_EXTRA_V}.tar.gz
+	https://github.com/jprichardson/node-jsonfile/archive/${NODE_JSONFILE_V}.tar.gz -> node-jsonfile-${NODE_JSONFILE_V}.tar.gz
+	https://github.com/jprichardson/node-klaw/archive/${NODE_KLAW_V}.tar.gz -> node-klaw-${NODE_KLAW_V}.tar.gz
+	https://github.com/isaacs/node-graceful-fs/archive/v${NODE_GRACEFUL_FS_V}.tar.gz -> node-graceful-fs-${NODE_GRACEFUL_FS_V}.tar.gz
+	https://github.com/isaacs/rimraf/archive/v${RIMRAF_V}.tar.gz -> node-rimraf-${RIMRAF_V}.tar.gz
+	https://github.com/tbranyen/combyne/archive/${COMBYNE_V}.tar.gz -> node-combyne-${COMBYNE_V}.tar.gz
+	https://github.com/beautify-web/js-beautify/archive/v${JS_BEAUTIFY_V}.tar.gz -> node-js-beautify-${JS_BEAUTIFY_V}.tar.gz
+"
+
+BINMODS="
+		cached-run-in-this-context
+		node-ctags
+		git-utils
+		nodegit
+		keyboard-layout
+		node-nslog
+		node-oniguruma
+		node-pathwatcher
+		node-runas
+		node-keytar
+		scrollbar-style
+		node-spellchecker
+		marker-index
+"
+
 RESTRICT="mirror"
 LICENSE="MIT"
 SLOT="0"
-KEYWORDS="~amd64 ~x86"
+KEYWORDS="~amd64"
 IUSE=""
 
 DEPEND="
+	!dev-util/apm
 	${PYTHON_DEPS}
-	net-libs/nodejs[npm]
-	media-fonts/inconsolata
-	gnome-base/gconf
-	x11-libs/gtk+:2
-	gnome-base/libgnome-keyring
-	x11-libs/libnotify
-	x11-libs/libXtst
-	dev-libs/nss
-	media-libs/alsa-lib
+	>=net-libs/nodejs-5.9.0:=[npm]
+	>=app-text/hunspell-1.3.3:=
+	=dev-libs/libgit2-0.23*:=[ssh]
+	>=dev-libs/oniguruma-5.9.5:=
+	>=dev-util/ctags-5.8
+	dev-util/electron:0/36
 "
 RDEPEND="${DEPEND}"
 
+S="${WORKDIR}/${PN}-${MY_PV}"
+
 pkg_setup() {
 	python-any-r1_pkg_setup
-	npm config set python $PYTHON
 }
 
-src_prepare(){
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
+}
+
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+package_dir() {
+	local binmod="${1}" binmod_v
+	eval binmod_v=\${$(echo ${binmod//-/_}_V | tr '[:lower:]' '[:upper:]')}
```

Comment:
First `$(echo)` call is extraneous. Please use the bash Here-String `<<<` operator. For instance, this:
```
$ export bar=foo
$ echo $bar | sed -e 's/foo/baz/;'
baz
```
becomes this:
```
$ sed -e 's/foo/baz/;' <<< $bar
baz
```